### PR TITLE
Heedls 418 administrator edit details

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-418

### Description
Adds a new page to edit admin roles, making use of new checkboxes and radios view components. ~~There are still some outstanding questions about permissions for the page to be resolved since the requirements seem to state this page is accessible to more people than its parent page. Have implemented with the same access for now.~~ Resolved the permissions issues, is correct as is implemented with CentreManager only (not Centre Admin). Updated the JIRA ticket to match.

### Screenshots
image
image
image

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by livzorn — APPROVED** [review]
No issues that I can find!

**Review by ibrahimmunir14 — CHANGES_REQUESTED** [review]

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Administrator/EditRolesViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    public class EditRolesViewModel
+    {
+        public readonly CheckboxListItemViewModel[] Checkboxes =
+        {
+            new CheckboxListItemViewModel(
+                nameof(IsCentreAdmin),
+                "Centre administrator",
+                "Manage delegates, courses and course groups. Enrol users on courses. View reports."
+            ),
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Administrator/EditRolesViewModel.cs:20]: I think it's more confusing to have improperly named properties/parameters, and properties we would never use the value of (Value would always be overwritten in the view component logic). Maybe I could move one of them from the namespace?

I can't make CheckboxesItemViewModel private like that, it needs to be accessed from both the view component Invoke and the model. I don't think the logic in the Invoke can be moved to the constructor of the CheckboxesItemViewModel in a sensible way.

**Comment by AlexJacksonDS** [comment]
Not really sure what I can do about the page structure issues you've raised. The fieldset labels match the NHS styling for the respective components. So does the dropdown.

They need to have labels, so the only real solution would be to have duplicate headers and make the label smaller than the NHS components:
image

**Comment by ibrahimmunir14** [comment]
> Not really sure what I can do about the page structure issues you've raised. The fieldset labels match the NHS styling for the respective components. So does the dropdown.
> 
> They need to have labels, so the only real solution would be to have duplicate headers and make the label smaller than the NHS components:

Ah. Well, to me this makes more sense structurally but duplicate headers look bad. Can we ask HEE?

**Comment by AlexJacksonDS** [comment]
> Ah. Well, to me this makes more sense structurally but duplicate headers look bad. Can we ask HEE?

I'll do it on Monday when everyone is back

**Comment by AlexJacksonDS** [comment]
Confirmed page structure changes with Kevin, updated the screenshots in the description.

**Review by ibrahimmunir14 — APPROVED** [review]
LGTM

**Pull request merged**